R/helper.R

```diff
@@ -19,3 +19,46 @@ get_by_id = function(xs, id) {
   }
   return(NULL)
 }
+
+
+#' @description calculate pareto front of a matrix
+#'   also works for >2D pareto front
+#' @example data = rbind(c(1, 0.25, 0.75, 0), c(0, 0.25, 0.75, 1))
+#'   calculate_pareto_front(data)
+#'   data = rbind(c(1,1,0,1), c(1,1,0,0), c(1,0,1,0), c(1,1,1,1))
+#'   calculate_pareto_front(data, maximize = FALSE)
+calculate_pareto_front = function(points, maximize = TRUE) {
+
+  assert_matrix(points, types = "numeric")
+  assert_logical(maximize)
```

Comment:
so minimize may have either length 1 or length (ncol(points))? or (nrow(points))? or really any length? I don't know, but should probably be in the assert here.

Remember you can make alternative assertions with
```
assert(
  check_logical(maximize, len = 1),
  check_logical(maximize, len = 10)
)
```
maybe there are even better ways of doing this.